File: opensource/data/src/main/java/com/sos/scheduler/engine/data/job/TaskEvent.java
package com.sos.scheduler.engine.data.job;

import com.sos.scheduler.engine.data.event.AbstractEvent;
import com.sos.scheduler.engine.data.folder.JobPath;

public abstract class TaskEvent extends AbstractEvent {
    private final TaskId id;
    private final JobPath jobPath;

    protected TaskEvent(TaskId id, JobPath jobPath) {
        this.jobPath = jobPath;
        this.id = id;
    }

    public final TaskId getId() {
        return id;
    }

    public final JobPath getJobPath() {
        return jobPath;
    }

    @Override public String toString() {
        return super.toString() +" "+ id +" ("+ jobPath +")";
    }
}

File: opensource/data/src/main/scala/com/sos/scheduler/engine/data/job/TaskEvent.scala
package com.sos.scheduler.engine.data.job

import com.sos.scheduler.engine.data.event.{KeyedEvent, Event}
import com.sos.scheduler.engine.data.folder.JobPath

trait TaskEvent extends KeyedEvent {
  type Key = TaskId
  final def key = taskId

  def taskId: TaskId
  def jobPath: JobPath
}
